### PR TITLE
Add required Debian dependencies to allow docker builds on the arm platform

### PR DESCRIPTION
**Pull request by maquis196**
arm platform

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
* [x] Pull request includes a [sign off](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#sign-off)
* [x] Code style is correct (run the [linters](https://github.com/matrix-org/synapse/blob/master/CONTRIBUTING.md#code-style))

### CODE REVIEW

**Comment by maquis196** [comment]
rebased + fixed merge conflict with my recent docker healthcheck PR

**Comment by clokep** [comment]
The tests failing are from #8167, merging in develop should fix them.

**Comment by clokep** [comment]
Just trying to explain *why* some of these packages are needed:
* libffi-dev required by [cffi](https://cffi.readthedocs.io/en/latest/installation.html)
* libjpeg-dev, zlib1g-dev, libjpeg62-turbo required by [Pillow](https://pillow.readthedocs.io/en/stable/installation.html#external-libraries)
* libssl-dev required by [cryptography](https://cryptography.io/en/latest/installation/#building-cryptography-on-linux)
* libxml++2.6-dev, libxslt1-dev -- I think these are both required by xmlsec1 via [PySAML2](https://pysaml2.readthedocs.io/en/latest/install.html)

**Comment by clokep** [comment]
I wonder if this should be adding webp like #7791 did too. We might have regressed support for that in Docker.

**Comment by maquis196** [comment]
> I wonder if this should be adding webp like #7791 did too. We might have regressed support for that in Docker.

Im surprised that didnt survive the conversion, yeah, it was added originally, let me check package names and ill add to this pr.

File: changelog.d/8144.docker [hunk 1]
```diff
@@ -0,0 +1 @@
+Added extra Debian devel packages to allow docker builds on the arm platform. Contributed by @maquis196.
```
Comment (richvdh) [inline changelog.d/8144.docker:1]: this is possibly better expressed as:

```suggestion
Fix builds of the Docker image on non-x86 platforms.
```

**Review by richvdh — APPROVED** [review]
lgtm otherwise

**Comment by mattcen** [comment]
@maquis196 Thanks for sorting this; I got bitten by this when trying to build on my RPi, and hadn't gotten around to sending through a PR of this myself, but it was on my to-do list!

**Pull request merged**